### DISCUSSION TITLE
Make `security.model.domain` accept undefined fields

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#225] [Ideas]
Right now the `security.model.domain` forces every field to be defined with either a null value or a string. This is inconvenient to use because you must write out every field, even if you aren't using all the fields for the domain.

**EXAMPLE**

```typescript
security: {
  project: {
    domain: {
      userId: null,
      roleCode: null,
      projectMemberId: null,
      projectId: "id",
    },
    permissions: {
      CREATE_PROJECT: PERMISSION.CREATE_ONLY,
      DELETE_PROJECT: PERMISSION.DELETE_ONLY,
      UPDATE_PROJECT: PERMISSION.UPDATE_ONLY,
    },
    defaultPermissions: PERMISSION.READ_ONLY
  },
}
```

However, this forced null declaration doesn't seem necessary, as if you pass in the domains with an `<any>` cast, the security middleware still works.

**STILL WORKS**

```typescript
security: {
  project: {
    domain: <any>{
      projectId: "id",
    },
    permissions: {
      CREATE_PROJECT: PERMISSION.CREATE_ONLY,
      DELETE_PROJECT: PERMISSION.DELETE_ONLY,
      UPDATE_PROJECT: PERMISSION.UPDATE_ONLY,
    },
    defaultPermissions: PERMISSION.READ_ONLY
  },
}
```

I think Typetta should remove the null requirement and instead make the fields optional.

**Comment by edobrb** [comment]
I was pretty convinced that null was needed in some cases, but I've checked I didn't see anything. I think it's due to previous versions. Thanks for the report. On version 1.6.4  it's fixed.
